src/SDK/SmartDetectorsSDK/AlertPresentation/AzureResourceManagerRequest.cs

```diff
@@ -0,0 +1,123 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="AzureResourceManagerRequest.cs" company="Microsoft Corporation">
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// </copyright>
+//-----------------------------------------------------------------------
+
+namespace Microsoft.Azure.Monitoring.SmartDetectors.AlertPresentation
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Xml;
+
+    /// <summary>
+    /// Base class for alert properties that get their data from the result of an ARM request.
+    /// Inheriting classes should add presentation properties to display the results of the ARM request.
+    /// </summary>
+    public abstract class AzureResourceManagerRequest
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AzureResourceManagerRequest"/> class.
+        /// </summary>
+        /// <param name="requestUri">The request's URI. This must be a relative URI that will be executed against the ARM endpoint.</param>
+        protected AzureResourceManagerRequest(Uri requestUri)
+        {
+            if (requestUri == null)
+            {
+                throw new ArgumentNullException(nameof(requestUri));
+            }
+
+            if (requestUri.IsAbsoluteUri)
+            {
+                throw new ArgumentException("The ARM request URI must be a relative URI", nameof(requestUri));
+            }
+
+            this.RequestUri = requestUri;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AzureResourceManagerRequest"/> class, based on the given resource and suffix.
+        /// </summary>
+        /// <param name="resource">The resource for which the ARM request should be made</param>
+        /// <param name="requestSuffix">An optional suffix for the request, which will appended to the resource's URI</param>
+        protected AzureResourceManagerRequest(ResourceIdentifier resource, string requestSuffix)
+        {
+            if (resource == null)
+            {
+                throw new ArgumentNullException(nameof(resource));
+            }
+
+            // Make sure that we'll be able to append the request suffix
+            if (string.IsNullOrEmpty(requestSuffix))
+            {
+                this.RequestUri = new Uri(resource.ToResourceId(), UriKind.Relative);
+            }
+            else if (requestSuffix[0] != '/' && requestSuffix[0] == '?')
```

Comment:
>if (requestSuffix[0] != '/' && requestSuffix[0] == '?') [](start = 17, length = 55)

Didn't you mean just:
        if (requestSuffix[0] != '/')
?

Reply:
actually was support to be != "?" - added tests
Thanks!

---
In reply to: [279190654](https://github.com/Azure/SmartDetectorsMonitoringAppliance/pull/61#discussion_r279190654) [](ancestors = 279190654)